packages/@react-spectrum/overlays/src/Popover.tsx

```diff
@@ -63,13 +70,83 @@ function Popover(props: PopoverProps, ref: RefObject<HTMLDivElement>) {
         <button {...dismissButtonProps} />
       </VisuallyHidden>
       {children}
-      {hideArrow ? null : <div className={classNames(styles, 'spectrum-Popover-tip')} {...arrowProps} data-testid="tip" />}
+      {hideArrow ? null : (
+        <Arrow arrowProps={arrowProps} direction={arrowPlacement[placement.split(' ')[0]]} />
+      )}
       <VisuallyHidden>
         <button {...dismissButtonProps} />
       </VisuallyHidden>
     </div>
   );
 }
 
+let ROOT_2 = Math.sqrt(2);
+
+function Arrow(props) {
+  let [size, setSize] = useState(20);
+  let [borderWidth, setBorderWidth] = useState(1);
+  let ref = useRef();
+  // get the css value for the tip size and divide it by 2 for this arrow implementation
+  useLayoutEffect(() => {
+    let spectrumTipWidth = getComputedStyle(ref.current)
```

Comment:
Maybe just call `getComputedStyle` once? Might be nice to actually reference `window` here too so its clear it's a global.